### DISCUSSION TITLE
Enable only selected cameras?

### DISCUSSION THREAD

**Discussion by TheWaterbug** [mrlt8/docker-wyze-bridge#1362] [Q&A]
I have two locations, my house and my parents' house, each with multiple Wyze cameras and a Blue Iris box running docker-wyze-bridge. All the Wyze cameras are under one account, which I manage.

Is there a way to restrict each container to see and attempt logins only selected cameras? e.g. only the cameras in the same physical location? There i is no need for the container at my place to see the cameras at my parents' place, nor vice versa.

I tried setting:

- NET_MODE=LAN

in my 

C:\Docker\WyzeBridge\docker-compose.yml

but when I restart the container (even restarted the whole PC!) I do not see that string when Inspect the container, and both containers are still seeing the cameras at the other location.

I also have a persistent S2S IPSec tunnel between our two houses, for various reasons, so I'm not even sure if NET_MODE=LAN would somehow work through that. The documentation just says "LAN", but doesn't specify if this means "same subnet" or ???

I have a theory that the Wyze cameras get unhappy when too many clients log into them, so I'm trying to restrict that to just the one wyze-bridge that is local to it, plus my very occasional login from the Wyze app to restart them as necessary 🤦🏻‍♂️

I also don't want 18 different cameras showing up in at localhost:5000 when I'm troubleshooting; I just want to see the cameras that are physical local to this container.

Thanks!

**Comment by mrlt8** [comment]
NET_MODE is kind of broken on the newer firmware as wyze seems to be using a newer version of the TUTK SDK which breaks the NAT punching and P2P/Relay connection modes. As a result, we're only able to connect to the cameras using the "LAN" connection mode. In other words, if you can ping the camera from the bridge, you can connect to it with "LAN" mode.

And yes, I believe the 2024 firmware also introduced some kind of connection limit for each resolution, so you can have 1 active connection to each resolution (one to the "HD/2k" and one to the "360p") and the camera will just kick off the other connection.

Will look into adding additional filtering options, but you could potentially filter by Mac address with something like this to only allow certain cameras:

```yaml
- FILTER_MACS=00:aA:22:33:44:55, Aa22334455bB
```

or add  `FILTER_BLOCK=true` to block the cameras listed in `FILTER_MACS`

**Reply by TheWaterbug** [reply]
Hmmm. I tried adding just 7 MACs to - FILTER_MACS=, but the cameras at the other location are still showing up.

Do I add - FILTER_MACS= to the Environment section of docker-compose.yml? Or to some other section?

Thanks!

**Reply by TheWaterbug** [reply]
Hmm. I've tried putting it here:

    ports:
            - 1935:1935 # RTMP
            - 8554:8554 # RTSP
            - 8888:8888 # HLS
            - 8889:8889 #WebRTC
            - 8189:8189/udp # WebRTC/ICE
            - 5000:5000 # WEB-UI
            - FILTER_MACS=d0:3f:27:45:70:cd, d0:3f:27:c6:e9:41, d0:3f:27:c4:c2:be, 2c:aa:8e:eb:dc:71, 80:48:2c:29:d7:de, 7c:78:b2:21:ea:d2, 7c:78:b2:21:ea:de


as well as here:

        environment:
            # [OPTIONAL] Credentials can be set in the WebUI
            # API Key and ID can be obtained from the wyze dev portal: 
            # https://developer-api-console.wyze.com/#/apikey/view
            - WYZE_EMAIL=<myEmailforWyze>
            - WYZE_PASSWORD=<myWyzePassword> 
            - API_ID=<myAPI-ID>
            - API_KEY= <my API key>
            # [OPTIONAL] IP Address of the host to enable WebRTC e.g.,:
            # - WB_IP=192.168.1.122
            # WebUI and Stream authentication:
            - WB_AUTH=False # Set to false to disable web and stream auth.
            # WB_USERNAME=
            # WB_PASSWORD=
            - NET_MODE=LAN
            - FILTER_MACS=d0:3f:27:45:70:cd, d0:3f:27:c6:e9:41, d0:3f:27:c4:c2:be, 2c:aa:8e:eb:dc:71, 80:48:2c:29:d7:de, 7c:78:b2:21:ea:d2, 7c:78:b2:21:ea:de


but it doesn't filter the cameras, and none of the MACs show up when I inspect the container:

		"Env": [
			"API_KEY= <my API key>",
			"WB_AUTH=False",
			"WYZE_EMAIL=<myEmailforWyze>",
			"WYZE_PASSWORD=<myWyzePassword>",
			"API_ID=<myAPI-ID>",
			"PATH=/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
			"LANG=C.UTF-8",
			"GPG_KEY=<myGPGKey>",
			"PYTHON_VERSION=3.12.6",
			"PYTHONUNBUFFERED=1",
			"FLASK_APP=frontend",
			"BUILD=2.10.3",
			"GITHUB_SHA=baee602505cb0dde40dda8ca4037357c8e50f49d"
		],

Do I have the syntax or placement wrong?

According to localhost:5000 I'm running docker-wyze-bridge v2.10.3 on Docker Desktop 4.37.1 (178610), so I have the latest.

Thanks!
